blog/assembly/contracts/main.ts

```diff
@@ -0,0 +1,53 @@
+import { Storage, callerHasWriteAccess, generateEvent } from '@massalabs/massa-as-sdk';
+import { Args } from '@massalabs/as-types';
+
+/**
+ * This function is meant to be called only one time: when the contract is deployed.
+ *
+ * @param _ - not used
+ */
+export function constructor(_: StaticArray<u8>): StaticArray<u8> {
+  // This line is important. It ensures that this function can't be called in the future.
+  // If you remove this check, someone could call your constructor function and reset your smart contract.
+  if (!callerHasWriteAccess()) {
+    return [];
+  }
+  const nBlogPosts = 0;
+  // Set the initial value of N_BLOG_POSTS to 0 in the storage of the contract
+  Storage.set("N_BLOG_POSTS", nBlogPosts.toString());
+  // This function is used to emit an event to the blockchain
+  generateEvent('Blog initiated');
+  return [];
+}
+
+export function post(_args: StaticArray<u8>): void {
+  const args = new Args(_args);
+  // Get the post from the arguments and check if it is valid (not null)
+  const post = args
+    .nextString()
+    .expect('Post argument is missing or invalid');
+  let numberOfPosts = parseInt(Storage.get("N_BLOG_POSTS"));
+  numberOfPosts += 1;
+  // Store the post in the storage of the contract with the key POST_postIndex
+  // The keys will have the following syntaxes: POST_1.0, POST_2.0, POST_3.0, etc.
```

Comment:
I'm not sure about the .0 in 1.0, 2.0 ... numberOfPosts is an integer so it should be POST_1, POST_2 ...

Reply:
It is indeed an integer but for some unusual reason numberOfPosts will always have the decimal part even when applying Math.floor() on it. 

![image](https://user-images.githubusercontent.com/92168447/231753270-4f8a61c9-f166-42a7-b605-533dc3014610.png)

Reply:
For isDeployingContract() to work I had to update to the latest massa-as-sdk version which resulted in my unit tests not working. When readingthe storage the returned value would always be NaN. I've addressed this issue to the run team.

Reply:
This is not a priority. You can safely ignore my comment

Reply:
Using [isDeployingContract](https://github.com/massalabs/massa-as-sdk/blob/main/assembly/std/context.ts#L46) now and fixed the decimal part with the help of @Ben-Rey